homeassistant/core.py

```diff
@@ -866,6 +876,10 @@ def async_fire(
             listeners = match_all_listeners + listeners
 
         event = Event(event_type, event_data, origin, time_fired, context)
+        context = event.context
+        if not context.origin_event:
+            context.origin_event = event
+            self._context_origin[context.id] = context
```

Comment:
I think we can drop this completely since logbook has access to the original context